chalk/shapes/path.py

```diff
@@ -1,122 +1,83 @@
 from __future__ import annotations
 
-import math
 from dataclasses import dataclass
-from typing import Any, List, Tuple, Union
+from typing import Any, Iterable, List, Tuple
 
 from chalk import transform as tx
 from chalk.envelope import Envelope
-from chalk.shapes.arc import ArcSegment
-from chalk.shapes.segment import Segment
 from chalk.shapes.shape import Shape
 from chalk.trace import Trace
+from chalk.trail import Located, Trail
 from chalk.transform import P2
-from chalk.types import Diagram
+from chalk.types import Diagram, Enveloped, Traceable
 from chalk.visitor import A, ShapeVisitor
 
 
 def make_path(segments: List[Tuple[float, float]]) -> Diagram:
     return Path.from_list_of_tuples(segments).stroke()
 
 
-SegmentLike = Union[Segment, ArcSegment]
-
-
 @dataclass
-class Path(Shape, tx.Transformable):
+class Path(Shape, Enveloped, Traceable):
     """Path class."""
 
-    segments: List[SegmentLike]
-
-    @classmethod
-    def from_point(cls, point: P2) -> Path:
-        return cls([Segment(point, point)])
-
-    @classmethod
-    def from_points(cls, points: List[P2]) -> Path:
-        return cls(
-            list([Segment(pt1, pt2) for pt1, pt2 in zip(points, points[1:])])
-        )
-
-    @classmethod
-    def from_pairs(cls, points: List[Tuple[P2, P2]]) -> Path:
-        return cls(list([Segment(pt1, pt2) for pt1, pt2 in points]))
-
-    @classmethod
-    def from_list_of_tuples(cls, coords: List[Tuple[float, float]]) -> Path:
-        points = list([P2(x, y) for x, y in coords])
-        return cls.from_points(points)
-
-    def points(self) -> List[P2]:
-        points = []
-        for i, seg in enumerate(self.segments):
-            if i == 0:
-                points.append(seg.p)
-            points.append(seg.q)
-        return points
+    loc_trails: List[Located]
 
+    # Monoid - compose
     @staticmethod
-    def hrule(length: float) -> Path:
-        return Path.from_list_of_tuples([(-length / 2, 0), (length / 2, 0)])
+    def empty() -> Path:
+        return Path([])
 
-    @staticmethod
-    def vrule(length: float) -> Path:
-        return Path.from_list_of_tuples([(0, -length / 2), (0, length / 2)])
+    def __add__(self, other: Path) -> Path:
```

Comment:
Should we define a `Monoid` abstract base class such that all its subclasses have defined `concat`, as we have discussed in #95?

Reply:
Yeah, I think we should do that, maybe in a next PR. For now repeating `mconcat` doesn't seem so bad.

Reply:
Yes, agreed; it's just a line of duplicated code.